### ISSUE TITLE
Training with more targets

### ISSUE THREAD

**Issue by alecmerdler** [vb000/Waveformer#3]
Thanks for creating this project, it is very interesting! I noticed that the list of targets to detect is hard-coded, is it possible to re-train the model for different targets?

**Comment by vb000** [comment]
@alecmerdler, yes it is possible to re-train the model for a different set of targets. It involves creating a new dataset and corresponding PyTorch's dataset class here:

https://github.com/vb000/Waveformer/blob/383a211ab5c6bf87ce525f219ad1b8d5de997cee/src/training/synthetic_dataset.py#L20

Any dataset that generates `(mixture, label, ground-truth)` tuple of tensors as items, can be used to train the model. The shapes of tensors are `1xT`, `1xL` and `1xT`, respectively, where T is the number of audio samples in time dimension and L is the total number of sound classes you're training the model with.

Once you have that, you could re-train the model by updating the following fields in the config.json:

https://github.com/vb000/Waveformer/blob/383a211ab5c6bf87ce525f219ad1b8d5de997cee/experiments/dcc_tf_ckpt_E256_10_D128_1/config.json#L5

and 

https://github.com/vb000/Waveformer/blob/383a211ab5c6bf87ce525f219ad1b8d5de997cee/experiments/dcc_tf_ckpt_E256_10_D128_1/config.json#L15-L35

`json` objects above are passed as `**kwargs` to the dataset class by the training script.

**Comment by alecmerdler** [comment]
@vb000 Awesome, thanks for the detailed response! I'll let you know after I try this and if I run into any issues.

**Comment by euihyeoklee** [comment]
@vb000 This is truly amazing. I also want to train a new model by adding a new target. If I understand correctly, it seems that when a new target is added, new files within the jams folder need to be created to generate new mixtures for new targets. Is my understanding correct? If so, could you guide me on how to create these files?
